auth/only_api.go

```diff
@@ -10,68 +9,24 @@ import (
 	"github.com/evergreen-ci/utility"
 	"github.com/mongodb/grip"
 	"github.com/mongodb/grip/message"
-	"gopkg.in/mgo.v2/bson"
-
 	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
 )
 
 // NewOnlyAPIUserManager creates a user manager for special users that can only
 // make API requests. Users cannot be created and must come from the database.
-func NewOnlyAPIUserManager(config *evergreen.OnlyAPIAuthConfig) (gimlet.UserManager, error) {
-	validUsers := make([]gimlet.User, 0, len(config.Users))
-	validIDs := make([]string, 0, len(config.Users))
-
-	catcher := grip.NewBasicCatcher()
-	for _, u := range config.Users {
-		// Synchronize API-only users with those in the database.
-		checkUser, err := user.FindOneById(u.Username)
-		if err != nil {
-			catcher.Wrapf(err, "could not check whether user '%s' exists in the DB already", u.Username)
-			continue
-		}
-		if checkUser != nil && !checkUser.OnlyAPI {
-			catcher.Errorf("cannot use API-only user '%s' who is not marked as API-only", u.Username)
-			continue
-		}
-
-		_, err = db.Upsert(user.Collection,
-			bson.M{user.IdKey: u.Username, user.OnlyAPIKey: true},
-			bson.M{
-				"$set": bson.M{
-					user.APIKeyKey:  u.Key,
-					user.RolesKey:   u.Roles,
-					user.OnlyAPIKey: true,
-				},
-				"$setOnInsert": bson.M{
-					user.IdKey: u.Username,
-				},
-			},
-		)
-		if err != nil {
-			catcher.Wrapf(err, "could not upsert API-only user '%s'", u.Username)
-			continue
-		}
-
-		dbUser := &user.DBUser{
-			Id:          u.Username,
-			APIKey:      u.Key,
-			SystemRoles: u.Roles,
-			OnlyAPI:     true,
-		}
-
-		validUsers = append(validUsers, dbUser)
-		validIDs = append(validIDs, u.Username)
+func NewOnlyAPIUserManager() (gimlet.UserManager, error) {
+	validUsers := []gimlet.User{}
+	validIDs := []string{}
+	users, err := user.FindServiceUsers()
```

Comment:
I actually meant that it's not necessary to call `FindServiceUsers` at all (L52-62 and L73-77) because you can just assume that a bot will have `only_api: true` in the DB.

Reply:
I told myself I would never do this, but alas I committed the change but never pushed. Should be fixed now